pos_frequency_pipeline.py

```diff
@@ -13,15 +18,45 @@ class EmptyFileError(Exception):
 
 class POSFrequencyPipeline:
     def __init__(self, corpus_manager: CorpusManager):
-        pass
+        self.corpus_manager = corpus_manager
 
     def run(self):
-        pass
+        for article in self.corpus_manager.get_articles().values():
+            single_tagged_path = article.get_file_path('single_tagged')
+            with open(single_tagged_path, 'r', encoding='utf-8') as file:
+                text = file.read()
+            if not text:
+                raise EmptyFileError
+            tokens = text.split()
+            pos_frequencies = {}
+            pos_pattern = re.compile('[A-Z]+')
```

Comment:
use groups in pattern and make it more specific for a tag so that it will not be mistaken by any uppercase letters